app/controllers/lambda_callback/document_proof_result_controller.rb

```diff
@@ -1,9 +1,14 @@
 module LambdaCallback
   class DocumentProofResultController < AuthTokenController
     def create
-      dcs = DocumentCaptureSession.new
-      dcs.result_id = result_id_parameter
-      dcs.store_proofing_result(document_result_parameter)
+      EncryptedRedisStructStorage.store(
+        ProofingDocumentCaptureSessionResult.new(
+          id: result_id_parameter,
+          pii: document_result_parameter[:pii_from_doc],
+          result: document_result_parameter.except(:pii_from_doc),
+        ),
+        expires_in: AppConfig.env.async_wait_timeout_seconds.to_i,
+      )
```

Comment:
The `DocumentCaptureSession` methods that store stuff in redis do some weird things, like generate a new `result_id` each time (which is not what we want) and have separate methods to store PII and the result.... so I figured I could do exactly what I wanted by writing it like this

Reply:
the `DocumentCaptureSession#store_proofing_result` method was aimed at this case. It's storing a result, and doesn't generate a new `result_id`.

The issue I think is that the non-doc auth proofing implementations use the same classes, but they have different needs. The non-doc auth proofing uses the PII field as inputs and parameters, but it's part of the result here. I think it may be potentially confusing and cause issues in the future to use the fields in divergent ways. Should they be using different structs?

Reply:
But if we used `store_pii_result` first, it would wipe the result_id out and make a new one....?